doc/design/cluster_train/submit-job.md

```diff
@@ -0,0 +1,115 @@
+- *NOTE*: The first stage, we only implement submit the PaddlePaddle job in `paddle.job.dist_train()`
+
+You can call `paddle.job.dist_train` and provide distributed training configuration as the parameters:
+```python
+paddle.job.dist_train(
+    trainer=paddle.trainer.SGD(...,
+                              paddle.updater.Adam(...)),
+    reader=reader,
+    paddle_job=PaddleJob(
+      job_name="quickstart",
+      use_gpu=True,
+      cpu_num=4,
+      gpu_num=2,
+      memory="1G"
+      input=/quickstart/input,
+      output=/quickstart/output,
+      base_image="paddlepaddle/paddle:0.10.rc2")
+```
+
+The pseudo code of `paddle.job.dist_train` is as follows:
+```python
+  def dist_train(trainer, reader, num_passes=1, event_handler=None, feeding=None, paddle_job=None):
+    # if the code is running on cloud, set PADDLE_ON_CLOUD=YES
+    if os.getenv("PADDLE_ON_CLOUD", "NO") == "NO":
+      #submit the paddle job
+      paddle_job.submit()
+    else:
+      #start the training
+      trainer.train(reader, num_passes, event_handler, feeding)
+```
+
+parameter | required | default | explain
+  --- | --- | --- | ---
+job_name|YES||you should special a uniq job name which in a namespace
+trainer_package|YES|| entry point for startup trainer process
+input| YES || input directory on distributed file system
+output|YES|| output directory on distributed file system
+pservers|YES|| parameter server process count
+base-image|YES||PaddlePaddle production Docker image
+memory|YES|| limits for memory
+use_gpu|NO|false| whether use GPU
+cpu_num|NO|1| if `use_gpu=false`, this parameter is required
+gpu_num|NO|1| if `use_gpu=true`, this parameter is required
+
+- Startup Parameter Server and Trainer Jobs
+  - Deploy parameter server job, it's a Kubernetes ReplicaSet.
+  - Deploy trainer job, it's a Kubernetes Job.
+
+# Job Server
+
+- RESTful API
+
+  Job server provides a RESTful HTTP API, receive the trainer package and display
+  PaddlePaddle job related informations.
+  - `POST   /v1/package` receive the trainer package and save them on CephFS
+  - `POST   /v1/trainer/job` submit a trainer job
+  - `GET    /v1/jobs/` list all jobs
+  - `GET    /v1/jobs/<job-name>` the status of a job
+  - `DELETE /v1/jobs/<job-name>` delete a job
+  - `GET    /v1/version` job server version
+
+- Build Runtime Docker Image on Kubernetes
+
+  `paddle.job.dist_train` will upload the trainer package to Job Server and then save them on the distributed filesystem, and then start up a job for building the runtime Docker image, Parameter Server and Trainer will use this runtime Docker image.
+
+  There are some benefits for building runtime Docker image on JobServer:
+  - **Docker in Docker** should mount `docker.sock` in the container and set `--privileged`, if the code running in a kubernetes pod, it's not safe.
+  - Users only need to upload the training package files, does not dependency docker engine, docker registry.
```

Comment:
does not dependency docker engine, docker registry. -> does not need to install docker engine, docker registry as dependencies.